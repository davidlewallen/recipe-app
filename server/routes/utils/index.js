const isAuthenticated = (req, res, next) => {
  if (req.isAuthenticated()) {
    next();
  } else {
    res.send(401);
  }
}

module.exports = {
  isAuthenticated,
};
