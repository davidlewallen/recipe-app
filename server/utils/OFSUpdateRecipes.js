const fs = require('fs');
const URLParse = require('url-parse');

const server = require('../db');
const RecipeModel = require('../models/recipe');
const { stripWebsite } = require('../controllers/websiteRules');

const updateRecipes = async () => {
  if (process.env.NODE_ENV !== 'production') {
    process.env.NODE_ENV = 'dev';
    const envResults = fs.readFileSync('../../.env', 'utf8').split('\n');
    envResults.forEach(env => {
      const splitEnv = env.split('=');
      process.env[splitEnv[0]] = splitEnv[1];
    });
  }

  try {
    await server.start();
    const recipes = await RecipeModel.find({});

    for (const recipe of recipes) {
      const parsedURL = URLParse(recipe.url.href);
      const strippedWebsite = await stripWebsite(parsedURL);
      await RecipeModel.findByIdAndUpdate(recipe._id, strippedWebsite);
    }
    process.exit(0);
  } catch (err) {
    console.log('err', err);
  }
};

module.exports = updateRecipes;

require('make-runnable');
