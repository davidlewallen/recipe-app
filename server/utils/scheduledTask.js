const axios = require('axios');

const updateNPWebsites = require('./updateNPWebsites');

const updateNPWebsitesTask = () => {
  const interval = process.env.NODE_ENV === 'dev' ? 10000 : 18000000;
  setInterval(() => updateNPWebsites.checkProcessableWebsites(), interval);
};

const noSleep = () => {
  if (process.env.NODE_ENV === 'prod') {
    setInterval(() => axios.get('http://www.mysavedrecipes.com'), 5000);
  }
};

const runTasks = () => {
  updateNPWebsitesTask();
  noSleep();
};

module.exports = { runTasks };
