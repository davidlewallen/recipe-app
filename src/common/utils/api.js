import axios from 'axios';

const API_ROOT = '/api';
const API_RECIPE = `${API_ROOT}/recipe`;
const API_ACCOUNT = `${API_ROOT}/account`;

export const Recipe = {
  getRecipes: () => axios.get(`${API_RECIPE}`),
  submitRecipe: encodedURI => axios.post(`${API_RECIPE}/submit/${encodedURI}`),
  deleteRecipe: recipeId => axios.delete(`${API_RECIPE}/delete/${recipeId}`),
};

export const Account = {
  login: body => axios.post(`${API_ACCOUNT}/login`, body),
  register: body => axios.post(`${API_ACCOUNT}/register`, body),
  logout: () => axios.get(`${API_ACCOUNT}/logout`),
  auth: () => axios.get(`${API_ACCOUNT}/auth`),
  getUser: () => axios.get(`${API_ACCOUNT}/user`),
  verify: (id, key) => axios.get(`${API_ACCOUNT}/verify?id=${id}&key=${key}`),
  resendVerification: id => axios.get(`${API_ACCOUNT}/verify/resend?id=${id}`),
};

export const Utils = {
  getAcceptedWebsites: () => axios.get(`${API_ROOT}/approved`),
};
