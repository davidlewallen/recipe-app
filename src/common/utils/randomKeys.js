// @flow
import uuid from 'uuid/v5';

const NAMESPACE = 'f829e458-5757-54ac-b4b7-598f44c2c9b5';

const genKey = (value: String): String => uuid(value.toString(), NAMESPACE);

export default genKey;
