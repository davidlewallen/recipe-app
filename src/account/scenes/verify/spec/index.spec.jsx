import React from 'react';
import { shallow } from 'enzyme';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';

import Verify from '..';

import { Account } from '../../../../common/utils/api';

const mock = new MockAdapter(axios);

describe('Verify', () => {
  let wrapper = null;
  let instance = null;
  let mockProps = {};

  beforeEach(() => {
    mockProps = {
      history: { replace: jest.fn() },
      userId: '1',
      verificationKey: 'test',
    };

    mock
      .onGet(
        Account.endpoints.verify(mockProps.userId, mockProps.verificationKey)
      )
      .reply(200);

    wrapper = shallow(<Verify {...mockProps} />);
    instance = wrapper.instance();
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('should call verifyEmail', () => {
      const spy = jest.spyOn(instance, 'verifyEmail');

      instance.componentDidMount();

      expect(spy).toHaveBeenCalled();
    });
  });

  describe('verifyEmail', () => {
    it('should set state.verificationState to "verified" if successful', async () => {
      wrapper.setState({ verificationState: 'checking' });

      await instance.verifyEmail();

      expect(wrapper.state('verificationState')).toEqual('verified');
    });

    it('should set state.verificationState to "resend" if status === 400 && verificationExpired is true', async () => {
      mock
        .onGet(
          Account.endpoints.verify(mockProps.userId, mockProps.verificationKey)
        )
        .reply(400, { verificationExpired: true });

      wrapper.setState({ verificationState: 'checking' });

      await instance.verifyEmail();

      expect(wrapper.state('verificationState')).toEqual('resend');
    });

    it('should set state.verificationState to "resend" if status === 400 && nonMatchingKey is true', async () => {
      mock
        .onGet(
          Account.endpoints.verify(mockProps.userId, mockProps.verificationKey)
        )
        .reply(400, { nonMatchingKey: true });

      wrapper.setState({ verificationState: 'checking' });

      await instance.verifyEmail();

      expect(wrapper.state('verificationState')).toEqual('nonMatching');
    });

    it('should redirect to "/dashboard" after 5 seconds if successful', async () => {
      jest.useFakeTimers();

      await instance.verifyEmail();

      jest.advanceTimersByTime(6000);

      expect(mockProps.history.replace).toHaveBeenCalledWith('/dashboard');
    });
  });

  describe('handleResendVerification', () => {
    it('should set verificiationState to "resent"', async () => {
      mock
        .onGet(Account.endpoints.resendVerification(mockProps.userId))
        .reply(200);

      await instance.handleResendVerification();

      expect(wrapper.state('verificationState')).toEqual('resent');
    });
  });
});
