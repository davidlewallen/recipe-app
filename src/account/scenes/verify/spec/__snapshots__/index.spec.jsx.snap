// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify should render 1`] = `
<VerifyEmail
  handleResendVerification={[Function]}
  verificationState="checking"
/>
`;
